File: prelim_analysis/seq_lengths.py
# SEQ METRICS: Confirms/denies the uniformity of seq lengths across datasets
from pandas import read_csv as pdr
import seaborn as sns
from math import sqrt
from matplotlib import pyplot as plt
metrics = pdr("~/Documents/amphibian_meta_project/meta_analysis/" +
        "qiime_analyses/seq-metrics/meta_bioinf_metrics.csv")
metrics['Seq_SE'] = metrics['Seq_SD']/sqrt(len(metrics['Seq_SD']))

# Plot
g = sns.FacetGrid(metrics, size=5)
g.map(plt.errorbar, 'Dataset', 'Mean_Seq_Lngth', 'Seq_SE',
        marker='o')
g.set_xticklabels(["Ellison et al., 2019",
        "Prado-Irwin et al., 2017",
        "Ellison et al., 2018",
        "Bird et al., 2018"])
g.ylabel('Mean Sequence Length')
plt.rcParams["axes.labelsize"] = 15
plt.title('Similarity of Sequence Lenths', fontsize = 20)
plt.show()

File: prelim_analysis/seq_lengths.R
# SEQ METRICS: Confirms/denies the uniformity of seq lengths across datasets
setwd("~/Documents/amphibian_meta_project/meta_analysis/qiime_analyses/seq-metrics")
pcks <- c("ggplot2", "wesanderson")
sapply(pcks, require, character.only = T)
metrics <- read.csv("meta_bioinf_metrics.csv")

# Plot
ggplot(metrics, aes(x = Dataset, y = Features)) +
    geom_boxplot() +
    xlab("Dataset") +
    ylab("Mean Sequence Length") +
    scale_x_discrete(breaks = c("Bird_et_al._2018",
                                "Ellison_et_al._2018",
                                "Ellison_et_al._2019",
                                "Prado-Irwin_et_al._2017"),
                     labels = c("Cal Salamanders Data",
                                "Mexico-Guatemala Data",
                                "Sierra Frogs Data",
                                "Ensatina Data")) +
    theme_bw() +
    theme(legend.position = "none",
          axis.title = element_text(size = 16, family = "Georgia"),
          axis.text.x = element_text(size = 12, family = "Georgia"),
          panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(),
          panel.border = element_blank(),
          axis.line = element_line(colour = "black", size = 0.25))
